examples/medplum-provider/src/pages/encounter/EncounterChart.tsx

```diff
@@ -1,13 +1,102 @@
-import { Textarea, Select, Button, Text, Paper, Stack, Group, Box } from '@mantine/core';
-import { Task } from '@medplum/fhirtypes';
-import { useSearchResources } from '@medplum/react';
+import { Select, Button, Text, Stack, Group, Box, Card } from '@mantine/core';
+import { Questionnaire, QuestionnaireResponse, Reference, Task } from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum } from '@medplum/react';
 import { useParams } from 'react-router-dom';
 import { TaskQuestionnaireForm } from '../components/TaskQuestionnaireForm';
 import { SimpleTask } from '../components/SimpleTask';
+import { useEffect, useMemo, useState } from 'react';
+import { showNotification } from '@mantine/notifications';
+import { getReferenceString, normalizeErrorString } from '@medplum/core';
+import { IconCircleOff } from '@tabler/icons-react';
 
 export const EncounterChart = (): JSX.Element => {
   const { encounterId } = useParams();
-  const [tasks] = useSearchResources('Task', `encounter=Encounter/${encounterId}`);
+  const medplum = useMedplum();
+  const [tasks, setTasks] = useState<Task[]>([]);
+  const [questionnaireResponse, setQuestionnaireResponse] = useState<QuestionnaireResponse | undefined>(undefined);
+  const [chartQuestionnaire, setChartQuestionnaire] = useState<Reference<Questionnaire> | undefined>(undefined);
+  const [chartQuestionnaireResponse, setChartQuestionnaireResponse] = useState<
+    Reference<QuestionnaireResponse> | undefined
+  >(undefined);
+  const questionnaireReference = 'Questionnaire/0194903a-74f5-72e9-9063-e38b8276a855';
+
+  useEffect(() => {
+    const fetchTasks = async (): Promise<void> => {
+      const result = await medplum.searchResources('Task', `encounter=Encounter/${encounterId}`);
+      setTasks(result);
+    };
+
+    fetchTasks().catch((err) => {
+      showNotification({
+        color: 'red',
+        icon: <IconCircleOff />,
+        title: 'Error',
+        message: normalizeErrorString(err),
+      });
+    });
+  }, [medplum, encounterId]);
+
+  const { filteredTasks, chartNoteTasks } = useMemo(() => {
+    if (!tasks) {
+      return { filteredTasks: [] as Task[], chartNoteTasks: [] as Task[] };
+    }
+
+    return tasks.reduce(
+      (acc, task) => {
+        if (
+          Array.isArray(task.input) &&
+          task.input.length > 0 &&
+          task.input[0]?.valueReference?.reference === questionnaireReference
```

Comment:
Is it possible that there are multiple tasks with the same questionnaire as an input? If so, we should probably change how we are deciding how to set the `chartQuestionnaireResponse`, since it is possible there are multiple. 

If not, we might want to use some other method than `.reduce` and exit early from the loop once we find our one response